app/models/friendship.rb

```diff
@@ -0,0 +1,11 @@
+class Friendship < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :friend, :class_name => "User"
+  validates_uniqueness_of :friend, scope: :user
+  validate :check_user
+  def check_user
```

Comment:
I'm finding it difficult to implement flash errors to get the check user error to render, so I might have to hand over to someone else.